java/arcs/core/crdt/CrdtSet.kt

```diff
@@ -19,7 +19,17 @@ import arcs.core.crdt.CrdtChange.Operations
 import arcs.core.crdt.CrdtSet.Data
 import arcs.core.data.util.ReferencablePrimitive
 
-/** A [CrdtModel] capable of managing a set of items [T]. */
+/**
+ * A [CrdtModel] capable of managing a set of items [T].
+ *
+ * The implementation is based on the optimized OR-Set as described in:
+ *
+ * Annette Bieniusa, Marek Zawirski, Nuno Preguiça, Marc Shapiro,
+ * Carlos Baquero, Valter Balegas, Sérgio Duarte,
+ * "An Optimized Conflict-free Replicated Set" (2012)
+ *
+ * https://arxiv.org/abs/1210.3368
```

Comment:
The code appears to match the implementation described in this paper, but I could definitely use a fact-checker here.

Reply:
Very nice, I highly recommend putting these kinds of comments with links to external references.